### PR TITLE
feat: sitemap

### LINKED ISSUE #91: Improve SEO
Read this [guide](https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#optimizing-your-docs-for-search-engine-spiders) and implement any of the possible steps.

For example, try adding a robot.txt file according to the [instructions](https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#use-a-robots-txt-file).

### PR DESCRIPTION
**Pull request by florian-hoenicke**

### CODE REVIEW

File: html_extra/robots.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+User-agent: *
+Disallow:
+sitemap: https://my-site.com/docs/sitemap.xml
```
Comment (florian-hoenicke) [inline html_extra/robots.txt]: needs to be refactored to have multiple sitemaps for older versions

File: conf.py [hunk 2]
```diff
@@ -42,6 +42,10 @@
 html_theme = 'sphinx_rtd_theme'
 
 base_url = '/'
+html_baseurl = 'https://docs.jina.ai'
+sitemap_url_scheme = '{link}'
+sitemap_locales = [None]
```
Comment (florian-hoenicke) [inline conf.py:47]: For now, we only support one local. In the future, we can add more languages.

**Comment by florian-hoenicke** [comment]
`https://docs.jina.ai/.github/pull_request_template.html` is in the sitemap.xml
 Also, `https://docs.jina.ai` is missing.

**Comment by florian-hoenicke** [comment]
>https://docs.jina.ai/.github/pull_request_template.html is in the sitemap.xml
Also, https://docs.jina.ai is missing

fixed - changed it to DirectoryHTMLBuilder (dirhtml)

**Comment by florian-hoenicke** [comment]
For some reason netlify still thinks it should use `make html`. 
Where do we need to configure it to use `make dirhtml`?
https://app.netlify.com/sites/jina-docs/deploys/603ac0ad59d6f60007b98a09

File: html_extra/robots.txt (hunk 1 above)
Comment (florian-hoenicke) [reply html_extra/robots.txt]: What we could do is running `make dirhtml` for all previous builds and create a sitemap retrospectively for each.
Also, to not add the sitemaps manually to the sitemap index, we should use our `cd.yml` script to add the sitemaps to the robots.txt (https://developers.google.com/search/docs/advanced/sitemaps/large-sitemaps?hl=en&visit_id=637500524312622086-433016932&rd=1)

**Comment by bwanglzu** [comment]
we need to test if `robot.txt` can be sync to `gh-pages` branch

**Comment by bwanglzu** [comment]
> For some reason netlify still thinks it should use make html.
Where do we need to configure it to use make dirhtml?
https://app.netlify.com/sites/jina-docs/deploys/603ac0ad59d6f60007b98a09

We need to config in netlify I think, cc @Roshanjossey

**Comment by florian-hoenicke** [comment]
> we need to test if robots.txt can be sync to gh-pages branch
Yes, the `robots.txt` is put into the `_build/dirhtml` folder and therefore will be pushed to the `gh-pages`.

Everything works when I run it locally.
For future tickets, there are three things that can be refactored:

1. remove `robots.txt` files for older versions.

2. add sitemaps to the `robots.txt` automatically for new versions:
User-agent: *
Disallow:
sitemap: https://docs.jina.ai/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.10/sitemap.xml

3. create sitemaps for older versions retrospectively.
User-agent: *
Disallow:
sitemap: https://docs.jina.ai/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.4/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.3/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.2/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.1/sitemap.xml
sitemap: https://docs.jina.ai/v1.0.0/sitemap.xml

**Comment by bwanglzu** [comment]
@florian-hoenicke it seems works on netlify, wondering what's the purpose for renaming `html` to `dirhtml`?

**Comment by florian-hoenicke** [comment]
> @florian-hoenicke it seems works on netlify, wondering what's the purpose for renaming `html` to `dirhtml`?

It is for generating links as       
`https://docs.jina.ai/chapters/core/setup`
instead of   
`https://docs.jina.ai/chapters/core/setup/index.html`

or 
`https://docs.jina.ai`
instead of 
`https://docs.jina.ai/index.html`
        
Running with dirhtml it uses the `DirectoryHTMLBuilder`https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/dirhtml.html

**Review by bwanglzu — APPROVED** [review]
great work @florian-hoenicke ! let's see the build result.

cc @FionnD @ThePfarrer

**Comment by FionnD** [comment]
> > we need to test if robots.txt can be sync to gh-pages branch
> > Yes, the `robots.txt` is put into the `_build/dirhtml` folder and therefore will be pushed to the `gh-pages`.
> 
> Everything works when I run it locally.
> For future tickets, there are three things that can be refactored:
> 
> 1. remove `robots.txt` files for older versions.
> 2. add sitemaps to the `robots.txt` automatically for new versions:
>    User-agent: *
>    Disallow:
>    sitemap: https://docs.jina.ai/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.10/sitemap.xml
> 3. create sitemaps for older versions retrospectively.
>    User-agent: *
>    Disallow:
>    sitemap: https://docs.jina.ai/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.4/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.3/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.2/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.1/sitemap.xml
>    sitemap: https://docs.jina.ai/v1.0.0/sitemap.xml

@florian-hoenicke can you add these issues to the repo please? 🙏

**Comment by florian-hoenicke** [comment]
> @florian-hoenicke can you add these issues to the repo please? 🙏

done

**Pull request merged**